### ISSUE TITLE
Spring Boot 3.0 Support

### ISSUE THREAD

**Issue by codeconsole** [grails/grails-core#12794]
### Feature description

Spring Boot 3.0 has been released with a JDK 17 baseline.

https://spring.io/blog/2022/11/24/spring-boot-3-0-goes-ga

Currently, it is possible to build and run Grails with JDK 17, but the most breaking change has been a switch to Jakarta EE 9 Baseline and the new `jakarta` namespace which should be done in parallel with upgrading to Groovy 4.0 https://github.com/grails/grails-core/issues/12373

`javax.servlet.http.HttpServletRequest` is now `jakarta.servlet.http.HttpServletRequest`

**Comment by davidkron** [comment]
Yes please, everything below Spring Boot 3, JDK 17, Jakarta EE9, Groovy 4 would be a disappointment for the next major version of Grails.

**Comment by puneetbehl** [comment]
The Grails Foundation Technology Advisory Board has decided that Grails 6 will require minimum Java 11 and we won't be able to support Spring Boot 3. We are planning to support Spring Boot 3 with Grails 7 which might be released middle of 2023.

**Comment by fernando88to** [comment]
Please if possible improve the coverage feature support.

https://github.com/grails/grails-core/issues/11318

**Comment by codeconsole** [comment]
@puneetbehl  Are we at least getting Groovy 4 in Grails 6?  Groovy 4 for has been around for a year now...
https://repo.maven.apache.org/maven2/org/apache/groovy/groovy-all/

**Comment by puneetbehl** [comment]
@fernando88to Definitely, https://github.com/grails/grails-core/issues/11318 is already planned and will be released as a part of Grails 5.3.0 in next week or so.

@codeconsole There are some challenges related to support Groovy 4 in combination with other tools such as Gradle. We have already spent a good amount of time for that and will try to see if that can be done. So far Groovy 4 is also scheduled for Grails 7 but I am going to give it another try and will be in a better state afterwards to see if that's possible with Grails 6.

**Comment by davebrown1975** [comment]
One more here eagerly awaiting the fix for #11318 to address compliance matters in our app.

**Comment by SimonWoidig** [comment]
Any updates?

**Comment by SunilVishwakarma** [comment]
> The Grails Foundation Technology Advisory Board has decided that Grails 6 will require minimum Java 11 and we won't be able to support Spring Boot 3. We are planning to support Spring Boot 3 with Grails 7 which might be released middle of 2023.

@puneetbehl  Any update on this, as when the Grails 7 will be released to support Spring Boot 3, since Spring Boot 2 is EOL soon.

**Comment by SimonWoidig** [comment]
> > The Grails Foundation Technology Advisory Board has decided that Grails 6 will require minimum Java 11 and we won't be able to support Spring Boot 3. We are planning to support Spring Boot 3 with Grails 7 which might be released middle of 2023.
> 
> @puneetbehl Any update on this, as when the Grails 7 will be released to support Spring Boot 3, since Spring Boot 2 is EOL soon.

That's what worries us aswell. We have multiple components written with pure Spring Boot and this component (also the largest one) is still on the old version of Spring Boot. We are fearing the coming CVEs and we are considering ditching Grails.

**Comment by puneetbehl** [comment]
The release cadence for Grails major release is an year. So the Grails 7 will be released around mid of this year.

**Comment by SunilVishwakarma** [comment]
> The release cadence for Grails major release is an year. So the Grails 7 will be released around mid of this year.

Thanks for the update.

**Comment by dk185173** [comment]
Hope this gets materialised sooner.

**Comment by nithinyama** [comment]
> The Grails Foundation Technology Advisory Board has decided that Grails 6 will require minimum Java 11 and we won't be able to support Spring Boot 3. We are planning to support Spring Boot 3 with Grails 7 which might be released middle of 2023.

Hi. I want to follow up on Grails 7 release. As per the quote above, says mid of 2023, but i don't see in Grails official website.  
Sorry I just saw the questions previous threads mentioned that Grails will come in mid of this year.  Is that still valid ? @puneetbehl

**Comment by puneetbehl** [comment]
Currently, we don't have a precise date, but we are actively working on releasing Grails 7 in July as part of our plan.

**Comment by Chintanjariwala** [comment]
Hello. I want to follow up on Grails 7 release. As per the comment above, says Grails 7 releasing in July. Can you provide precise date on Grails 7 release @puneetbehl

**Comment by jamesfredley** [comment]
@Chintanjariwala you can see progress on https://github.com/orgs/grails/projects/3.

**Comment by aggaa-netapp** [comment]
What is the rough timeline for Grails 7 release please? Is it Q4 2024, Q1 2025 or even later? Thanks!

**Comment by jeffscottbrown** [comment]
> What is the rough timeline for Grails 7 release please? Is it Q4 2024, Q1 2025 or even later?

FYI... I don't anticipate it would be later.  The team [is targetting shipping M1 of Grails 8 in October 2025](https://github.com/grails/grails-core/issues/13757#issue-2589881216).  If the release of Grails 7 GA went much further out than Q1, that would start to get tight I think.

**Comment by jeffscottbrown** [comment]
@ajay-aggarwal FYI... You can keep an eye on Grails 7 progress at https://github.com/orgs/grails/projects/3.

**Comment by mindexpert7546** [comment]
How to implement the sso in grails 6.2.3

**Comment by jamesfredley** [comment]
Grails 7.0.0-M1 was released in December with Spring Boot 3.4.x:  https://grails.org/blog/2024-12-23-grails-7-m1.html.  See blog post for more details on milestones and releases.
